src/rttable.c

```diff
@@ -349,6 +397,12 @@ int insertRoute(uint32_t group, int ifx) {
         // Register the VIF activity for the aging routine
         BIT_SET(croute->ageVifBits, ifx);
 
+        // Update dwnstrHosts
+        if(findDownstreamSlot(croute, src) == -1){
+            int free = findFreeDownstreamSlot(croute);
```

Comment:
Do not use identifier `free`. It may cause conflicts with standard `free()` function. This function returns slot number/index, so `idx`, `slot` or `slotNum` or `slotNumber` or `slotIdx`. would be better name.